pkg/pipeline/sink/m3u8/reader.go

```diff
@@ -0,0 +1,62 @@
+package m3u8
+
+import (
+	"os"
+	"strconv"
+	"strings"
+	"time"
+)
+
+type Playlist struct {
+	Version        int
+	MediaType      string
+	TargetDuration int
+	Segments       []*Segment
+	Closed         bool
+}
+
+type Segment struct {
+	ProgramDateTime time.Time
+	Duration        float64
+	Filename        string
+}
+
+func ReadPlaylist(filename string) (*Playlist, error) {
+	b, err := os.ReadFile(filename)
+	if err != nil {
+		return nil, err
+	}
+
+	lines := strings.Split(string(b), "\n")
+	version, _ := strconv.Atoi(strings.Split(lines[1], ":")[1])
+	mediaType := strings.Split(lines[2], ":")[1]
+	targetDuration, _ := strconv.Atoi(strings.Split(lines[5], ":")[1])
+
+	p := &Playlist{
+		Version:        version,
+		MediaType:      mediaType,
+		TargetDuration: targetDuration,
+		Segments:       make([]*Segment, 0),
+	}
+
+	for i := 6; i < len(lines)-3; i += 3 {
+		startTime, _ := time.Parse(time.RFC3339Nano, strings.SplitN(lines[i], ":", 2)[1])
```

Comment:
This also assumes the PROGRAM-DATE-TIME tag is present, which may not be true in the general case.